### DISCUSSION TITLE
file not found: /tmp/weconnect.token

### DISCUSSION THREAD

**Discussion by costabel** [tillsteinbach/VWsFriend#91] [Q&A] [answered]
Trying to run the suggested command

docker-compose --env-file ./myconfig.env up

after long successful database startup messages, I get an error message (repeated in a loop): 

vwsfriend_1   | 2021-10-27T13:10:58+0000:INFO:vwsfriend_base:vwsfriend 0.9.3 (using WeConnect-python 0.21.5)
vwsfriend_1   | 2021-10-27T13:10:58+0000:INFO:weconnect:Could not use token from file /tmp/weconnect.token ([Errno 2] No such file or directory: '/tmp/weconnect.token')
vwsfriend_1   | The futex facility returned an unexpected error code.
vwsfriend_1   | Aborted (core dumped)
vwsfriend_vwsfriend_1 exited with code 134

This is independent of the fact whether there exists such a file. At first, there was no such file. Then I ran (successfully) weconnect-cli, which created such a file. The error message from the docker-compose command is the same in both situations.

**Comment by tillsteinbach** [comment, marked as answer]
The "No such file" message is fine. It just tells that there was no token file and will create one.

Regarding "The futex facility returned an unexpected error code." I have never seen that before. Can you give more information about the platform you are running on?

**Comment by costabel** [comment]
This is on a Raspberry Pi 3, under raspbian buster.

**Reply by linusnyren** [reply]
I'm having the same problem with a rpi 3b+ and raspbian buster.
I've updated and upgraded raspbian, docker and docker-compose.
Still no luck.
There is a huge scene for homekit integrations on rpi with homebridge, so i think this would be really great if solved.
I'm not experienced in python unfortunately so not that much of help

**Reply by tillsteinbach** [reply]
@costabel the problem should be fixed in 0.9.5

**Reply by costabel** [reply]
I am sorry for the late reaction (was traveling), but after updating everything to the latest version, in particular vwsfriend to version 0.9.5, I am still getting *exactly* the same error message. It still says

vwsfriend_1   | 2021-11-08T16:29:02+0000:INFO:vwsfriend_base:vwsfriend 0.9.3 (using WeConnect-python 0.21.5)
vwsfriend_1   | 2021-11-08T16:29:02+0000:INFO:weconnect:Could not use token from file /tmp/weconnect.token ([Errno 2] No such file or directory: '/tmp/weconnect.token')
vwsfriend_1   | The futex facility returned an unexpected error code.

Why does it still mention 0.9.3? Is there some cache that needs purging? I uninstalled and reinstalled everything, but to no avail.

**Reply by tillsteinbach** [reply]
Hey! If it says 0.9.3 it still is 0.9.3
Did you update with `docker-compose pull`?

**Reply by costabel** [reply]
I see that my email answer is not visible here. So just to complete this thread, here it is:
I had updated using "pip3 install --upgrade vwsfriend". After this, it looked like version 0.9.5 was installed. The bug was still there, however.
Then I used "docker-compose pull" as suggested. This made the bug disappear.
It works nicely, now at version 0.10.1. There are still lots of warnings in the logs, though. One says to report it as bug, which I will do in another thread.

**Reply by tillsteinbach** [reply]
Good to hear that it is working now. Will look into your other bug report as soon as possible! Thanks!
